scripts/components/Explore.js

```diff
@@ -1,39 +1,53 @@
 'use strict';
 
 import React from 'react';
-import LinkedStateMixin from 'react/lib/LinkedStateMixin';
-import { Navigation } from 'react-router';
 
-const Explore = React.createClass({
-  mixins: [LinkedStateMixin, Navigation],
+const DEFAULT_LOGIN = 'gaearon';
+const parseFullName = (params) => {
+  if (!params.login) {
+    return DEFAULT_LOGIN;
+  }
+  return params.login + (params.name ? '/' + params.name : '');
+};
+
+// TODO: update input when URL changes
+class Explore extends React.Component {
 
-  getInitialState() {
-    return {
-      loginOrRepo: 'gaearon'
+  constructor(props) {
+    this.state = {
+      loginOrRepo: parseFullName(props.params)
     };
-  },
+  }
 
   render() {
     return (
       <div className='Explore'>
         <p>Type a username or repo full name and hit 'Go':</p>
-        <input valueLink={this.linkState('loginOrRepo')}
-               onKeyUp={this.handleKeyUp} />
-        <button onClick={this.handleGoClick}>Go!</button>
+        <input
+          ref='loginOrRepo'
+          onKeyUp={this.handleKeyUp.bind(this)}
+          defaultValue={this.state.loginOrRepo} />
+        <button onClick={this.handleGoClick.bind(this)}>Go!</button>
         <p>Code on <a href='https://github.com/gaearon/flux-react-router-example' target='_blank'>Github</a>.</p>
       </div>
     );
-  },
+  }
 
   handleKeyUp(e) {
     if (e.keyCode === 13) {
-      this.handleGoClick();
+      this.handleGoClick.call(this);
```

Comment:
Why `.call(this)`? It should be exactly the same as just calling `this.handleGoClick()`.